tinygrad/ops.py

```diff
@@ -1081,7 +1083,10 @@ def sint_to_uop(x:sint, dtype:DType=dtypes.int) -> UOp: return UOp.const(dtype,
 symbolic = symbolic_simple+PatternMatcher([
   # ** COMMUTATIVE flipping **
   (UPat(GroupOp.Commutative, name='x'), lambda x: x.replace(src=x.src[::-1]) if x.src[1].tuplize < x.src[0].tuplize else None),
-  # group like
+  # ** group stuff **
+  # flattening of ADD, MULL chains etc, generalization of (x1+x2) + (x3+x4) -> x1+x2+x3+x4
```

Comment:
typo `MULL`, also you just need `Associative` for this right? i know you are working on followups, but always good to keep individual ones standalone and minimum

Reply:
It does flip the ordering so it needs to be commutative too `(x1+x2)+x3` becomes `(x1+x3)+x2`